storage/src/mmr/journaled.rs

```diff
@@ -1,32 +1,57 @@
 //! An MMR backed by a fixed-item-length journal.
+//!
+//! A [`crate::journal`] is used to store all unpruned MMR nodes, and a [`crate::metadata`] store is
+//! used to preserve each current peak digest in case they would otherwise have been be pruned.
 
 use crate::journal::{
     fixed::{Config as JConfig, Journal},
     Error as JError,
 };
+use crate::metadata::{Config as MConfig, Metadata};
 use crate::mmr::{
-    iterator::PeakIterator,
+    iterator::{oldest_provable_pos, oldest_required_proof_pos, PeakIterator},
     mem::Mmr as MemMmr,
     verification::{Proof, Storage},
     Error,
 };
+use bytes::Bytes;
 use commonware_cryptography::Hasher;
-use commonware_runtime::{Blob, Metrics, Storage as RStorage};
-use commonware_utils::SizedSerialize;
-use futures::future::try_join_all;
-use tracing::warn;
+use commonware_runtime::{Blob, Clock, Metrics, Storage as RStorage};
+use commonware_utils::array::U64;
+use tracing::{error, warn};
+
+/// Configuration for a journal-backed MMR.
+#[derive(Clone)]
+pub struct Config {
+    /// The name of the `commonware-runtime::Storage` storage partition used for the journal storing
+    /// the MMR nodes.
+    pub journal_partition: String,
+
+    /// The name of the `commonware-runtime::Storage` storage partition used for the metadata
+    /// containing the MMR's current peaks.
+    pub metadata_partition: String,
+
+    /// The maximum number of items to store in each blob in the backing journal.
```

Comment:
nit: this is "the number of items" not "maximum" right?

Reply:
no it's the max because the most recent blob may not be full.